pkg/gvk/gvk.go

```diff
@@ -153,6 +223,59 @@ func collectPreferredResources(discovery discovery.DiscoveryInterface) ([]*metav
 	return resources, nil
 }
 
+// collectPreferredResources collects all preferred namespaced scoped apiResources from the cluster
+func preferredCRDResource(discovery discovery.DiscoveryInterface) ([]*metav1.APIResourceList, error) {
```

Comment:
NIT: rename this to `collectPreferredCRDResources`? I think the doc comment needs an update as well

Reply:
@alaypatel07 Makes sense -- although the singular "resource" was deliberate, since we will be returning exactly one resource in the slice -- the preferred GV for "customresourcedefinitions" -- should I call it `collectPreferredCRDResource` to indicate that, or should I make it plural like all the rest since it returns a slice, even though it will have a length of one?

Reply:
`collectPreferredCRDResource ` works